server: Add rangefeed metrics

miretskiy opened the pull request:
Add rangefeed related metrics to keep track of the number of actively running rangefeeds on the server.

Epic: CRDB-26372
Release note: None

Commit:
server: Add rangefeed metrics

Add rangefeed related metrics to keep track of the
number of actively running rangefeeds on the server.

Epic: CRDB-26372
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108062)
<!-- Reviewable:end -->

Commit:
kvcoord: Correctly add rangefeed metrics to metric registry

Annotate `DistSenderRangeFeedMetrics` as `metric.Struct`
in order to register those metrics with metrics registry.

Issue: None
Release note: None

erikgrinaker commented on pkg/server/node.go:
```diff
@@ -195,6 +195,18 @@ This metric is thus not an indicator of KV health.`,
 		Unit:        metric.Unit_BYTES,
 	}
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "stream.rangefeed.count",
```
How about an `rpc.streams.` prefix for consistency with other RPC metrics, and have both a counter and a gauge for each method?

We could also consider generating automatic metrics for all streaming methods, similarly to `rpc.method.%s.recv` for regular requests. But we can punt that as long as the metrics scheme is extensible.

erikgrinaker commented on pkg/server/node.go:
```diff
@@ -195,6 +195,30 @@ This metric is thus not an indicator of KV health.`,
+		Name:        "rpc.streams.rangefeed.count",
+		Help:        `Number of currently running RangeFeed streams`,
+		Measurement: "Count",
+		Unit:        metric.Unit_BYTES,
```
`Unit_COUNT`

miretskiy replied:
done.

erikgrinaker commented on pkg/server/node.go:
```diff
@@ -195,6 +195,30 @@ This metric is thus not an indicator of KV health.`,
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "rpc.streams.rangefeed.count",
+		Help:        `Number of currently running RangeFeed streams`,
+		Measurement: "Count",
```
Streams

erikgrinaker commented on pkg/server/node.go:
```diff
@@ -195,6 +195,30 @@ This metric is thus not an indicator of KV health.`,
 		Unit:        metric.Unit_BYTES,
 	}
+	metaActiveRangeFeed = metric.Metadata{
+		Name:        "rpc.streams.rangefeed.count",
```
Count seems like a misnomer, since this is a gauge not a counter. Active? Current?

miretskiy replied:
Ack.

miretskiy commented:
bors r+
